File: public/usage-examples/windows/clear_window-1-example.cpp
#include "splashkit.h"

int main()
{
    // open a window
    window wind = open_window("Colour Changer", 600, 200);

    while (!quit_requested())
    {
        // get user events
        process_events();

        // clear screen
        clear_window(wind, COLOR_WHITE);

        // Buttons that change the colour of the screen
        if (button("Red!", rectangle_from(75, 85, 100, 30)))
        {
            clear_window(wind, COLOR_RED);
            refresh_window(wind);
            delay(1000);
            continue;
        }
        if (button("Green!", rectangle_from(250, 85, 100, 30)))
        {
            clear_window(wind, COLOR_GREEN);
            refresh_window(wind);
            delay(1000);
            continue;
        }

        if (button("Blue!", rectangle_from(425, 85, 100, 30)))
        {
            clear_window(wind, COLOR_BLUE);
            refresh_window(wind);
            delay(1000);
            continue;
        }

        draw_interface();
        refresh_window(wind);
    }

    // close all open windows
    close_all_windows();
}

File: public/usage-examples/windows/clear_window-1-example.py
from splashkit import *

# open a window
wind = open_window("Colour Changer", 600, 200)

# main loop
while not quit_requested():
    # get user events
    process_events()
    
    # clear screen
    clear_window(wind, color_white())
    
    if button_at_position("Red!", rectangle_from(75, 85, 100, 30)):
        clear_window(wind, color_red())
        refresh_window(wind)
        delay(1000)
        continue
    
    if button_at_position("Green!", rectangle_from(250, 85, 100, 30)):
        clear_window(wind, color_green())
        refresh_window(wind)
        delay(1000)
        continue
    
    if button_at_position("Blue!", rectangle_from(425, 85, 100, 30)):
        clear_window(wind, color_blue())
        refresh_window(wind)
        delay(1000)
        continue
    
    # finally draw interface, then refresh screen
    draw_interface()
    refresh_window(wind)

# close all open windows
close_all_windows()